src/mcts/node.cc

```diff
@@ -555,11 +555,48 @@ V6TrainingData Node::GetV6TrainingData(
   std::fill(std::begin(result.probabilities), std::end(result.probabilities),
             -1);
   // Set moves probabilities according to their relative amount of visits.
+  // Compute Kullback-Leibler divergence in nats (between policy and visits).
+  float kld_sum = 0;
+  float max_p = -std::numeric_limits<float>::infinity();
+  std::vector<float> intermediate;
+  if (nneval) {
+    int last_idx = 0;
+    for (const auto& child : Edges()) {
+      auto nn_idx = child.edge()->GetMove().as_nn_index(transform);
+      float p = 0;
+      for (int i = 0; i < nneval->p.size(); i++) {
+        // Optimization: usually moves are stored in the same order as queried.
+        const auto& move = nneval->p[last_idx++];
+        if (last_idx == nneval->p.size()) last_idx = 0;
+        if (move.first == nn_idx) {
+          p = move.second;
+          break;
+        }
+      }
+      intermediate.emplace_back(p);
+      max_p = std::max(max_p, p);
+    }
+  }
+  float total = 0.0;
+  auto it = intermediate.begin();
   for (const auto& child : Edges()) {
-    result.probabilities[child.edge()->GetMove().as_nn_index(transform)] =
-        total_n > 0 ? child.GetN() / static_cast<float>(total_n) : 1;
+    auto nn_idx = child.edge()->GetMove().as_nn_index(transform);
+    float fracv = total_n > 0 ? child.GetN() / static_cast<float>(total_n) : 1;
+    if (fracv > 0 && nneval) {
+      float P = std::exp((*it - max_p) / temp);
```

Comment:
It isn't clear to me that temp should be applied in calculating the KLD.
The policy is trained to equal the visit distribution of training - without PST.  PST is used in training search, but that doesn't mean we should be measuring surprise of the network output vs its training target using it...

Reply:
Should we define a separate option for this?

Reply:
A separate option seems much more sensible than the current option - but not sure if we'd use it....

Reply:
In my opinion, the surprise should always be measured against the raw policies (PST=1.0) because that's what it is supposed to predict/be trained on. The purpose of PST in training is to counteract the sharpening of PUCT in RL in a specific way, giving a source of regularization.

Reply:
I can remove the temperature completely - I only added it when converting the original pr to get the policy from the cache, it was not there originally. If we have a need for it in the future it is easy to add back.